prisma/schema.prisma

```diff
@@ -35,13 +37,28 @@ model Shift {
   startTime        DateTime
   endTime          DateTime
   workerSlots      Int
+  blockedWorkers   BlockedWorker[]
 }
 
 model ShiftAssignment {
   shiftUuid  String
   workerUuid String
   shift      Shift  @relation(fields: [shiftUuid], references: [uuid])
   worker     Worker @relation(fields: [workerUuid], references: [uuid])
+  rating     Int?
 
   @@id([shiftUuid, workerUuid])
 }
+
+model BlockedWorker {
+  shiftUuid    String
+  workerUuid   String
+  facilityUuid String
+  shift        Shift              @relation(fields: [shiftUuid], references: [uuid])
+  worker       Worker             @relation(fields: [workerUuid], references: [uuid])
+  facility     HealthCareFacility @relation(fields: [facilityUuid], references: [uuid])
+  createdAt    DateTime
+  blockReason  String
+
+  @@id([facilityUuid, workerUuid])
+}
```

Comment:
- Consider adding indexes for facilityUuid and workerUuid fields to improve query performance.
- Consider adding updatedAt field to BlockedWorker model for better audit tracking